qmcpy/integrand/um_bridge_wrapper.py

```diff
@@ -0,0 +1,164 @@
+    DigitalNetB2 (DiscreteDistribution Object)
+        d               3
+        dvec            [0 1 2]
+        randomize       LMS_DS
+        graycode        0
+        entropy         7
+        spawn_key       ()
+    >>> _ = os.system('docker rm -f muqbp > /dev/null')
+    >>> class TestModel(umbridge.Model):
+    ...     def __init__(self):
+    ...         super().__init__("forward")
+    ...     def get_input_sizes(self, config):
+    ...         return [1,2,3]
+    ...     def get_output_sizes(self, config):
+    ...         return [3,2,1]
+    ...     def __call__(self, parameters, config):
+    ...         out0 = [parameters[2][0],sum(parameters[2][:2]),sum(parameters[2])]
+    ...         out1 = [parameters[1][0],sum(parameters[1])]
+    ...         out2 = [parameters[0]]
+    ...         return [out0,out1,out2]
+    ...     def supports_evaluate(self):
+    ...         return True
+    >>> my_model = TestModel()
+    >>> my_distribution = Uniform(
+    ...     sampler = DigitalNetB2(dimension=sum(my_model.get_input_sizes(config={})),seed=7),
+    ...     lower_bound = -1,
+    ...     upper_bound = 1)
+    >>> my_integrand = UMBridgeWrapper(my_distribution,my_model)
+    >>> my_solution,my_data = CubQMCNetG(my_integrand,abs_tol=5e-2).integrate()
+    >>> my_data
+    LDTransformData (AccumulateData Object)
+        solution        [-2.328e-10 -4.657e-10 -6.985e-10 -2.328e-10 -4.657e-10 -2.328e-10]
+        comb_bound_low  [-9.649e-06 -1.765e-04 -2.352e-04 -3.053e-07 -1.997e-05 -1.952e-05]
+        comb_bound_high [9.649e-06 1.765e-04 2.352e-04 3.048e-07 1.997e-05 1.952e-05]
+        comb_flags      [ True  True  True  True  True  True]
+        n_total         2^(10)
+        n               [1024. 1024. 1024. 1024. 1024. 1024.]
+        time_integrate  ...
+    CubQMCNetG (StoppingCriterion Object)
+        abs_tol         0.050
+        rel_tol         0
+        n_init          2^(10)
+        n_max           2^(35)
+    UMBridgeWrapper (Integrand Object)
+    Uniform (TrueMeasure Object)
+        lower_bound     -1
+        upper_bound     1
+    DigitalNetB2 (DiscreteDistribution Object)
+        d               6
+        dvec            [0 1 2 3 4 5]
+        randomize       LMS_DS
+        graycode        0
+        entropy         7
+        spawn_key       ()
+    >>> my_integrand.to_umbridge_out_sizes(my_solution)
+    [[-2.3283064365386963e-10, -4.656612873077393e-10, -6.984919309616089e-10], [-2.3283064365386963e-10, -4.656612873077393e-10], [-2.3283064365386963e-10]]
+    >>> my_integrand.to_umbridge_out_sizes(my_data.comb_bound_low)
+    [[-9.649316780269146e-06, -0.00017654551993473433, -0.00023524149401055183], [-3.0527962735504843e-07, -1.997367189687793e-05], [-1.9521190552040935e-05]]
+    >>> my_integrand.to_umbridge_out_sizes(my_data.comb_bound_high)
+    [[9.648851118981838e-06, 0.00017654458861215971, 0.0002352400970266899], [3.048139660677407e-07, 1.9972740574303316e-05], [1.9520724890753627e-05]]
```

Comment:
@linusseelinger here is a doctest showing QMCPy's compatibility with UM-Bridge input / output sizes.

Reply:
Awesome, looks great!